x-pack/plugins/ml/public/application/explorer/add_to_dashboard_control.tsx

```diff
@@ -0,0 +1,339 @@
+              return addedDashboards[item.id] === 'pending' ? (
+                <EuiLoadingSpinner size="m" />
+              ) : (
+                <EuiToolTip
+                  content={
+                    addedDashboards[item.id] === 'success' ? (
+                      <FormattedMessage
+                        id="xpack.ml.explorer.dashboardsTitle.successfullyAddedMessage"
+                        defaultMessage="Successfully added"
+                      />
+                    ) : (
+                      <FormattedMessage
+                        id="xpack.ml.explorer.dashboardsTable.quickAddLabel"
+                        defaultMessage="Quick add"
+                      />
+                    )
+                  }
+                >
+                  <EuiButtonIcon
+                    color="primary"
+                    onClick={addSwimlaneToDashboardCallback.bind(null, item)}
+                    iconType={
+                      addedDashboards[item.id] === 'success'
+                        ? 'checkInCircleFilled'
+                        : 'plusInCircle'
+                    }
+                    aria-label="Quick add"
+                    disabled={addedDashboards[item.id] === 'success'}
+                  />
+                </EuiToolTip>
+              );
+            },
+          },
+          {
+            name: i18n.translate('xpack.ml.explorer.dashboardsTable.editInDashboard', {
+              defaultMessage: 'Edit in dashboard',
+            }),
+            onClick: async (item) => {
+              // TODO generate an edit URL with predefined state instead (https://github.com/elastic/kibana/issues/69152)
+              await addSwimlaneToDashboardCallback(item);
+              setAddedDashboards({ ...addedDashboards, [item.id]: 'redirecting' });
+              await navigateToUrl(await dashboardService.getDashboardEditUrl(item.id));
+            },
+            type: 'icon',
+            icon: 'pencil',
+            enabled: (item) => !addedDashboards[item.id],
+          },
+        ],
+      },
+    ],
+    [addedDashboards, selectedSwimlanes]
+  );
+
+  const swimlaneTypeOptions = [
+    {
+      id: SWIMLANE_TYPE.OVERALL,
+      label: i18n.translate('xpack.ml.explorer.overallLabel', {
+        defaultMessage: 'Overall',
+      }),
+    },
+    {
+      id: SWIMLANE_TYPE.VIEW_BY,
+      label: i18n.translate('xpack.ml.explorer.viewByFieldLabel', {
+        defaultMessage: 'View by {viewByField}, up to {limit} rows',
+        values: { viewByField: viewBy, limit },
+      }),
+    },
+  ];
+
+  const redirectToDashboardInProgress = Object.entries(addedDashboards).find(
+    ([id, status]) => status === 'redirecting'
+  );
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={onClose}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage
+              id="xpack.ml.explorer.dashboardsTitle"
+              defaultMessage="Attach swimlanes to dashboards"
```

Comment:
Nit: Is `Attach` a verb Kibana generally uses for this type of action? Maybe `Add` is simpler and it's in line with the button text on the dashboard page itself.

Reply:
Changed in https://github.com/elastic/kibana/pull/68784/commits/5eb6b458ee51391f3d1c78455b5a39994683b6c7